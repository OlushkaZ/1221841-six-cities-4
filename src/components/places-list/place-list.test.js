import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import PlaceList from "./place-list.jsx";

const props = {
  offers: [
    {
      pictures: [`img/apartment-01.jpg`],
      premium: false,
      cost: 120,
      description: [`Good apartment`],
      type: `Apartment`,
      rating: 5,
      title: `Place cool`,
      bedrooms: 2,
      guests: 10,
      conveniences: [`Beautiful`],
      coordinates: [52.3909553943508, 4.85309666406198],
      owner: {
        avatar: `img/avatar-angelina.jpg`,
        name: `Clara`,
        pro: false,
      },
      id: 11,
      reviwes: [{}, {}],
    }, {
      pictures: [`img/apartment-04.jpg`],
      premium: true,
      cost: 780,
      description: [`Nice`],
      type: `Hotel`,
      rating: 1,
      title: `House in forest`,
      bedrooms: 22,
      guests: 100,
      conveniences: [`Tolter`],
      coordinates: [52.3909553943508, 4.85309666406198],
      owner: {
        avatar: `img/avatar-angelina.jpg`,
        name: `Poporo`,
        pro: true,
      },
      id: 67,
      reviwes: [{}, {}],
    }, {
      pictures: [`img/apartment-05.jpg`],
      premium: false,
      cost: 50,
      description: [`Good hotel`],
      type: `Apartment`,
      rating: 0,
      title: `Place cool`,
      bedrooms: 22,
      guests: 1,
      conveniences: [`TV`, `Toilet`],
      coordinates: [52.3909553943508, 4.85309666406198],
      owner: {
        avatar: `img/avatar-angelina.jpg`,
        name: `Feder`,
        pro: false,
      },
      id: 607,
      reviwes: [{}, {}],
    }],

  classes: {
    card: `cities__place-`,
    wrapper: `cities`,
    cards: `cities__places-`,
    map: `cities`,
  },
  onAdvertCardTitleClick: jest.fn(),
};

const mockStore = configureStore([]);

it(`Render PlaceList`, () => {
  const store = mockStore({
    onAdvertCardMouseOver: jest.fn(),
    onAdvertCardMouseOut: jest.fn(),
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <PlaceList {...props}/>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
