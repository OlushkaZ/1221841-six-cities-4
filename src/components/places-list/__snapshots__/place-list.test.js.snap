// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PlaceList 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            120
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <a
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          href=""
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </a>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <a
        href="/offer/89089"
        onClick={[Function]}
      >
        <h2
          className="place-card__name"
        >
          good rererer
        </h2>
      </a>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            12000
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <a
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          href=""
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </a>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "undefined%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <a
        href="/offer/89"
        onClick={[Function]}
      >
        <h2
          className="place-card__name"
        >
          good rererer
        </h2>
      </a>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-07.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            120
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <a
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          href=""
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </a>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <a
        href="/offer/8989"
        onClick={[Function]}
      >
        <h2
          className="place-card__name"
        >
          good rererer
        </h2>
      </a>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
</div>
`;
